test/hotel_system_test.rb

```diff
@@ -0,0 +1,259 @@
+				19.times do
+					@hotel.make_reservation(@range)
+				end
+
+				inval_range = Hotel::DateRange.new(Date.new(2019, 03, 03), Date.new(2019, 03, 10))
+
+				expect{@hotel.make_reservation(inval_range)}.must_raise ArgumentError
+			end
+		end
+	end
+
+	describe "#get_reservations_by_date" do
+
+		before do
+			@hotel = Hotel::HotelSystem.new
+
+			range = Hotel::DateRange.new(Date.new(2019, 12, 1), Date.new(2019, 12, 5))
+			range_two = Hotel::DateRange.new(Date.new(2019, 11, 27), Date.new(2019, 12, 2))
+			range_three = Hotel::DateRange.new(Date.new(2019, 12, 3), Date.new(2019, 12, 9))
+			@hotel.reservations << Hotel::Reservation.new(Hotel::Room.new(3), range)
+			@hotel.reservations << Hotel::Reservation.new(Hotel::Room.new(4), range_two)
+			@hotel.reservations << Hotel::Reservation.new(Hotel::Room.new(5), range_three)
+
+			searching = Date.new(2019, 12, 1)
+			@found_reservations = @hotel.get_reservations_by_date(searching)
+		end
+
+		it "returns an array of reservations" do
+			expect(@found_reservations).must_be_instance_of Array
+		end
+
+		it "returns an array the size of 2" do
+			expect(@found_reservations.length).must_equal 2
+		end
+
+	end
+
+	describe "#get_reservations_by_room" do
+
+		before do
+			@hotel = Hotel::HotelSystem.new
+
+			range = Hotel::DateRange.new(Date.new(2019, 12, 1), Date.new(2019, 12, 5))
+			range_two = Hotel::DateRange.new(Date.new(2019, 11, 27), Date.new(2019, 12, 1))
+			range_three = Hotel::DateRange.new(Date.new(2019, 12, 3), Date.new(2019, 12, 9))
+			res_one = Hotel::Reservation.new(@hotel.rooms[0], range)
+			res_two = Hotel::Reservation.new(@hotel.rooms[0], range_two)
+			res_three = Hotel::Reservation.new(@hotel.rooms[4], range_three)
+			@hotel.rooms[0].reservations << res_one
+			@hotel.rooms[0].reservations << res_two
+			@hotel.rooms[4].reservations << res_three
+		end
+
+		it "returns array length of 2" do
+			searching = Hotel::DateRange.new(Date.new(2019, 11, 26), Date.new(2019, 12, 5))
+			found_reservations = @hotel.get_reservations_by_room(@hotel.rooms[0], searching)
+
+			expect(found_reservations.length).must_equal 2
+		end
+
+	end
+
+	describe "#get_rooms_by_date" do
+
+		before do
+			@hotel = Hotel::HotelSystem.new
+
+			range = Hotel::DateRange.new(Date.new(2019, 12, 1), Date.new(2019, 12, 5))
+			range_two = Hotel::DateRange.new(Date.new(2019, 11, 27), Date.new(2019, 12, 2))
+			range_three = Hotel::DateRange.new(Date.new(2019, 12, 1), Date.new(2019, 12, 9))
+			@hotel.reservations << Hotel::Reservation.new(@hotel.rooms[0], range)
+			@hotel.reservations << Hotel::Reservation.new(@hotel.rooms[3], range_two)
+			@hotel.reservations << Hotel::Reservation.new(@hotel.rooms[4], range_three)
+
+			searching = Date.new(2019, 12, 1)
+			@available_rooms = @hotel.get_rooms_by_date(searching)
+		end
+
+		it "returns an array of 17 for available rooms on a date" do
+			expect(@available_rooms.length).must_equal 17
+		end
```

Comment:
I see you tested for this edge case above with `make_reservation`, but you should also test the edge case here of what happens if there are no available rooms.